json_serializable/lib/src/utils.dart

```diff
@@ -27,6 +27,62 @@ JsonSerializable valueForAnnotation(ConstantReader annotation) =>
 bool isEnum(DartType targetType) =>
     targetType is InterfaceType && targetType.element.isEnum;
 
+final _enumMapExpando = new Expando<Map<FieldElement, dynamic>>();
+
+/// If [targetType] is an enum, returns a [Map] of the [FieldElement] instances
+/// associated with the enum values mapped to the [String] values that represent
+/// the serialized output.
+///
+/// By default, the [String] value is just the name of the enum value.
+/// If the enum value is annotated with [JsonKey], then the `name` property is
+/// used if it's set and not `null`.
+///
+/// If [targetType] is not an enum, `null` is returned.
+Map<FieldElement, dynamic> enumFieldsMap(DartType targetType) {
+  if (targetType is InterfaceType && targetType.element.isEnum) {
+    var value = _enumMapExpando[targetType];
```

Comment:
nit: `return _enumMapExpando[targetType] ??= ...`?

Reply:
also, I would consider taking moving big anonymous closure out to a private method or even just a local variable so its easier to reason about what is happening here

Reply:
ack